Add Brazilian documents generation and documentation

lucasqueiroz opened the pull request:
This Pull Request closes the issue #1208.
For this generation, I used the #1382 checksum generator (I separated it into two methods so it can be reused, added documentation and an alias).
It generates a valid Brazilian ID based on SSP rules.

lucasqueiroz commented:
I noticed the tests are a bit messy. Maybe it could have a test for the checksum methods so it validates the algorithm and another one for the generator, validating it generates a random value with the correct amount of digits + format

lucasqueiroz commented:
I have refactored the tests so it doesn't require the checksum calculation inside the generator tests.

dainmiller commented:
Love this. Bumping to see if it can get merged - was going to make a similar change myself.

vbrazo commented:
@lucasqueiroz could you fix the merge issues? I'd like to include this PR in version 2. Thanks 👍

lucasqueiroz commented:
Sure, will take a look at it asap!

lucasqueiroz commented:
Hi @vbrazo, I have fixed the conflicts and added the RG generator to the ID documentation.

Please let me know if there is any change required.

Thanks!
